js/modules/k6/html/gen/gen_elements.go

```diff
@@ -0,0 +1,532 @@
+package main
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+// The ast parser populates this struct using the <ElemName>TagName const and <ElemName>Element struct
+type ElemInfo struct {
+	ConstName     string
+	StructName    string
+	PrtStructName string
+}
+
+// The signature for functions which inspect the ast nodes
+type ElemInfoCollector func(node ast.Node) ElemInfoCollector
+
+type ElemInfoCollectorState struct {
+	handler   ElemInfoCollector // The current function to check ast nodes
+	elemName  string            // Only valid when a TagName const or Element struct encountered and used as an index into elemInfos
+	elemInfos map[string]*ElemInfo
+}
+
+// Built from an entry in funcDefs by buildFuncDef(funcDef)
+type FuncDef struct {
+	ElemName   string
+	ElemMethod string
+	AttrMethod string
+	AttrName   string
+	ReturnType string
+	ReturnBody string
+	ReturnOpts []string
+}
+
+var funcDefs = []string{
+	"Href Download string",
+	"Href ReferrerPolicy enum=,no-referrer,no-referrer-when-downgrade,origin,origin-when-cross-origin,unsafe-url",
```

Comment:
I ended up going with option 1 using the struct. 

`[]struct {
	Elem, Method, Attr, TemplateType string
	Opts                             []string
}`


There's duplication between the Method and Attr fields of the struct because the method name doesn't always map directly to the name of a DOM attribute. The attribute is usually the method name lowercased but counter-examples are "Encoding" -> "enctype" and "AcceptCharset" ->  "accept-charset". 

If you'd like to get rid of the duplication then the Attr field of the struct can be dropped and the Method name can have an optional override like {"FormElement", "AcceptCharset=accept-charset", "string", nil}

It's pretty easy to do, just not sure if you'd prefer the explicit over brevity in this case.